### PR TITLE
Make Routes dynamic from mainSections

### PR DESCRIPTION
**Pull request by AndreaLandiArk**
Now to add an section is sufficient to add his component in the mainSections array with all the data needed to also set the menu item.

Updated the r3f package to fix the refresh issue.
WI : 15959

### CODE REVIEW

File: src/main.tsx [hunk 1]
```diff
@@ -20,27 +20,20 @@ import PlaygroundView from "./features/playground/playgroundView";
...
-          //Understand why with this method :params in path are not being recognized.
-          /*mainSections.map((x) =>
-          x.subsections.map((s) => (
-            <Route path={s.path} element={s.component({ a: dispatch })} />
-          ))
-        )*/
-        }
+        {mainSections.map((x) =>
+          x.subsections.map((s) =>
+            s.component ? <Route path={s.path} element={s.component()} /> : null
```
Comment (AndreaCuneo) [inline src/main.tsx:34]: the mainSections is a nested structure.

each mainSection has a path (x.path) and nested each subsection. 

you need a <Route> for each main section, and another Router for each subsection inside

I expect a route /main/jsonplaceholder

File: src/siteMap/mainSections.ts [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+import ConfigTableExampleView from "../features/configTable/configTableExample";
+import MovieTableView from "../features/paginatedTable/moviePage";
+import VideoGamesTableView from "../features/formExample/videoGamesPage";
 
-/*This is the Main Section ARRAY populate this to populate the main nav menu*/
+/*This is the Main Section ARRAY populate this to populate the main nav menu
+It is also used to create all the Routes for the router*/
 export const mainSections: MainSectionType[] = [
   {
     path: "/main",
```
Comment (AndreaCuneo) [inline src/siteMap/mainSections.ts:14]: this is not used in Route

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/main.tsx (hunk 1 above)
Comment (AndreaLandiArk) [reply src/main.tsx:34]: it is already a double map, we only need routes for subsections in this design. The main section is only to group them in the accordion in the SideBar. I am removing the path in the main sections

File: src/components/sideBar/menuItem/types.ts [hunk 3]
```diff
@@ -1,14 +1,13 @@
-import { ReactNode } from "react";
 import { IconType } from "react-icons";
-import { Action, AnyAction, Dispatch } from "redux";
+import { Dispatch } from "redux";
 
 interface SectionItem {
-  path: string;
+  path?: string;
```
Comment (AndreaCuneo) [inline src/components/sideBar/menuItem/types.ts:5]: making it optional doesn't mean you can Ignore it.

the mainSections.map(x => ...)

should use x.path if present.

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/siteMap/mainSections.ts [hunk 4]
```diff
@@ -1,48 +1,111 @@
...
+    subsections: [
+      {
+        path: "https://www.google.com",
+        label: "Google",
+        icon: FaExternalLinkAlt,
+        isInMenu: true,
+        isExternal: true,
+      },
+      {
+        path: "https://https://react.dev/",
```
Comment (AndreaCuneo) [inline src/siteMap/mainSections.ts:76]: ?

File: src/siteMap/mainSections.ts [hunk 5]
```diff
@@ -1,48 +1,111 @@
...
         isExternal: false,
+        component: VideoGamesTableView,
+      },
+    ],
+  },
+  {
+    label: "External Section",
+    subsections: [
+      {
+        path: "https://www.google.com",
```
Comment (AndreaCuneo) [inline src/siteMap/mainSections.ts:69]: externalUrl?

there is a dedicated property in the Type

File: src/main.tsx [hunk 6]
```diff
@@ -20,27 +20,22 @@ import PlaygroundView from "./features/playground/playgroundView";
...
-        {
-          //Understand why with this method :params in path are not being recognized.
-          /*mainSections.map((x) =>
-          x.subsections.map((s) => (
-            <Route path={s.path} element={s.component({ a: dispatch })} />
-          ))
-        )*/
-        }
+        {mainSections.map((x) =>
+          x.subsections.map((s) =>
```
Comment (AndreaCuneo) [inline src/main.tsx:33]: and what if 's' has further nestes subsections?

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/main.tsx (hunk 6 above)
Comment (AndreaLandiArk) [reply src/main.tsx:33]: would you support infinite with recursion in the project template? 
I was thinking that 2 were enough and more would be a project dedicated task if needed

File: src/siteMap/mainSections.ts [hunk 7]
```diff
@@ -1,48 +1,120 @@
...
+      },
+    ],
+  },
+  {
+    label: "Another Section",
+
+    subsections: [
+      {
+        label: "Sub Subsections",
+        hasSubsections: true,
```
Comment (AndreaCuneo) [inline src/siteMap/mainSections.ts:91]: use the presence of them in the array.

`if array has at least 1 element`

File: src/siteMap/mainSections.ts [hunk 8]
```diff
@@ -1,48 +1,120 @@
...
+        isInMenu: true,
+        isExternal: true,
+      },
+    ],
+  },
+  {
+    label: "Another Section",
+
+    subsections: [
+      {
```
Comment (AndreaCuneo) [inline src/siteMap/mainSections.ts:89]: subsection can have path. show example how to use them and route composition.

sections are useful to separate idential routes across... sections.

e.g.

/stuff 
    /list 
    /new

/otherstuff 
    /list 
    /new

**Review by AndreaCuneo — CHANGES_REQUESTED** [review]

File: src/siteMap/mainSections.ts (hunk 8 above)
Comment (AndreaLandiArk) [reply src/siteMap/mainSections.ts:89]: OK i used parent path to get all the page and component unique paths across sections

**Review by sean-sligo — APPROVED** [review]

**Pull request merged**